### ISSUE TITLE
Can't use buffer~ on Raspian Stretch

### ISSUE THREAD

**Issue by tommyecho** [porres/pd-cyclone#451]

**Comment by tommyecho** [comment]
Hello,

I’m now running puredata 49 on Raspian Stretch. When I run puredata, it prints “priority 6 scheduling failed, running at normal priority”. Then, when I try to run a simple test patch with cyclone/buffer, I get a series of error messages. I also get error messages when I open the cyclone/buffer~ help file. 

When the DSP is on, I get a series of messages saying "Warning: [buffer~]'s size not defined

When I ask it to find the last error, it says “last trackable error: no such array ‘o-tom_1003’” (the buffer has been given the name “tom”)

When I open the buffer help file, it says “$2: argument number out of range”

Running similar patches and the helpfile on my mac works just fine. I can't seem to figure out what's going on here, and have been struggling with it for a couple of weeks

When I ran dmesg, I get a lot of lines like this. I'm not sure if they're related or not:

[30392.414319] w1_master_driver w1_bus_master1: Family 0 for 00.124000000000.ba is not registered.
[30444.377823] w1_master_driver w1_bus_master1: Attaching one wire slave 00.924000000000 crc 36
[30444.383364] w1_master_driver w1_bus_master1: Family 0 for 00.924000000000.36 is not registered.

**Comment by porres** [comment]
> When I ask it to find the last error, it says “last trackable error: no such array ‘o-tom_1003’” (the buffer has been given the name “tom”)

I get a similar error when using pd 0.48

maybe for some crazy reason you dont't realkly have 0.49? 

I'm closing this issue because other people have confirmed on facebook that buffer~ works for them in the Raspberry Pi with Pd 0.49, so we don't really have a cyclone issue here.

And buffer~ is also just an abstraction, and the errors don't seem to be on cyclone objects used in the abstration. Hence, it would be for all practical terms, a Pd issue where Pd vanilla objects aren't behaving as expected on your system.

Buffer~ uses new features in the [soundfiler] object, added to Pd 0.49, so I assume this is where you have to look for. 

Maybe ask help on the Pd list to make sure you have the latest Pd version installed and if the Pd Vanilla objects are behaving as expected on your system.

**Comment by tommyecho** [comment]
I’ve verified a bunch that it’s .49. Are you running on linux? This is so difficult, I’ve been trying for about 80 hours


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
try using the read message in [soundfiler] without an array and see if it works as documented for pd 0.49

"If no array name is given, no samples are read but the info is provided anyway."

<img width="929" alt="Screen Shot 2019-07-31 at 14 21 45" src="https://user-images.githubusercontent.com/5360142/62233268-95be6b80-b39e-11e9-8040-9dfe7496c142.png">

**Comment by tommyecho** [comment]
Thanks for this help. It seems that that behavior is working fine with the soundfiler object. When I send it the read message, but remove the array name from the read message, the array window doesn’t change, but the info is sent out the right outlet to the print object.

I have succeeded in getting the buffer~ to read a soundfile. Still, when I turn on DSP, I get a bunch of messages saying the array size is not defined. It also gives a couple of messages saying the array can’t be found. Unfortunately, it can’t find the error for me. This is where I’ve been stuck for a long time now. It seems that the size messages and crop messages don’t get to where they need to go?


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
it helps if you share the patch that's not working for you. Best if it really isolates the object and what goes wrong with it...

Start with the help file examples. Can you tell me which of the examples in the help file will give you an error?

> It seems that the size messages and crop messages don’t get to where they need to go?
 
I know size/crop are not being used in the help files, so they're not what's causing error on it.

**Comment by porres** [comment]
We need to isolate this, really... if it's a cyclone related issue I'll reopen this, but again, I don't think it can be as none of the issues seem to come from cyclone objects used in the abstraction (such as "uzi", "mstosamps~", etc).

**Comment by tommyecho** [comment]
Thanks for helping me with this.

Ok, here’s a very simple patch that creates the behavior, and here are two different screen captures of error messages. The first (about buffer size) is what happens when I turn on dsp for this patch. The second (about $2) happens when I open the help file. This $2 message also happens when I load the buffer with a wav. file from disk. I can load the file and see it, but still can’t get DSP to work without freezing things.

I know that the mstosamps~ object works when the dsp is on. I’m not sure if whatever the problem is causes the setsize message to not make it’s way to it’s final destination, because things freeze up too much with DSP on. . . I have tested to make sure mstosamps works in a separate patch and helpfile, and it is fine. My suspicion is that there is something in the buffer~ subpatch that needs to be modified to run correctly within the linux raspberry pi, but I don’t really know. . . 







Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
screenshot helps, but I don't see them. Also, try to upload the patches in a zip format

**Comment by tommyecho** [comment]
I put the screenshots in this zip file. Maybe they’ll show up that way

The message for this patch is “Warning: [buffer~]’s size not defined” and “array: couldn’t find named array ’tom_1003’”

The error message that prints when I open the buffer~ help file (and when I load a file from disk into buffer) is “$2: argument number out of range"


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
i dont see any zip file to download

**Comment by tommyecho** [comment]
hmm I wonder if it’s because I’m sending from my mac?  I just uploaded the zip to google drive. Here’s the link:

https://drive.google.com/open?id=1a6jL1Fub_m6X6UecBuwEuRktACKv4oio


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by tommyecho** [comment]
One new behavior thing with that patch running on raspberry pi: if I read a file into the buffer, which works fine, it is deleted from the buffer when I turn on the dsp 


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by tommyecho** [comment]
Here’s something new. 

I set a trigger to do these three things in series: turn on DSP, bang a setsize message into the buffer~, and turn off dsp. This succeeds in changing the array size, which I confirmed by looking in the buffers subpatch. I still get the “size not defined”message though, and no improvements of any kind

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
please try to be more objective so  it's easier to debug this.

I downloaded your patch. what triggers the errors, just laoding the buffer object, not sending any other message to it?

**Comment by porres** [comment]
Like...

when you open the patch you sent me, it prints “Warning: [buffer~]’s size not defined” and “array: couldn’t find named array ’tom_1003’”

then if you you try to load a file from disk with the "read" message you get “$2: argument number out of range", right?

**Comment by porres** [comment]
the "$2: argument number out of range" error means that a Pd message with $2 expected a message with at least two elements but got a moessage with only one instead, like this 

<img width="194" alt="Screen Shot 2019-07-31 at 19 37 00" src="https://user-images.githubusercontent.com/5360142/62252930-9ec53200-b3ca-11e9-95b5-d8506ba868be.png">

The 'read' message in cyclone/buffer~ goes into [cyclone/fromsymbol @separator .] so it can get the file extension as the second element in that message. If this is not working, one possibility is that you're loading a file that does not have an extension somehow...

<img width="418" alt="Screen Shot 2019-07-31 at 19 41 20" src="https://user-images.githubusercontent.com/5360142/62253117-43477400-b3cb-11e9-84bc-8abccf33dc06.png">


Above, if I send hi.wav - it works. If I send just "hi", we get that error.

Maybe [cyclone/fromsymbol] is not working for raspberry pi? I don't know, you can test its help file.

Let me know...

you can also just test ands try something like this to see what is printed...

**Comment by tommyecho** [comment]
What you’re saying is almost correct. Nothing happens when I open the patch, but, when I turn on the dsp for the patch, the message about buffer size streams repeatedly in my print window. The message about the second variable appears whenever I open the help file. This same message also happens right at the moment when I load a wav.file from the disk into the buffer object.

I’m trying to be as objectively clear as I can in my description of this. Is this not clear? It seems it’s the same thing I just wrote. I’m doing my best. Thanks for your time

". . .here are two different screen captures of error messages. The first (about buffer size) is what happens when I turn on dsp for this patch. The second (about $2) happens when I open the help file. This $2 message also happens when I load the buffer with a wav. file from disk. “


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by tommyecho** [comment]
Thanks for this. I figured the $2 message meant that a list was not being filled with 2 elements. Unfortunately, I have failed thus far to find where this problem occurs. 

To be clear, I am successful in reading files from disk ( I said this in an earlier message). Still, when I turn on DSP, I get the error message about the file size not being defined. This message happens repeteadly and the system just freezes up.

Sorry that this is a hassle. I’m wondering if anyone is really using the buffer~ object on raspian? Everyone’s comments have been a bit unclear around this. I’m definitely a fan of objectivity and wish I could find some clearly sequenced documentation around this issue. My intention is to resolve this with the hopes that this will pave a way for others to deal with this situation when it arises, but perhaps I am just missing something that no one else will ever miss. At any rate, I appreciate your work in making the objects and understand if this is just something that won’t work.

Best,

Tom

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
it's just that I opened the example help patch and it all seemed a bit confusing, I had to go back and see it. 

The idea of uploading a test patch is that it's clear and quite objective to point out the error, with comment descriptions, and without anything else unnecessary (like cyclone/out~) and a bunch of messages that aren't being used. So, something like 

<img width="308" alt="Screen Shot 2019-07-31 at 20 05 58" src="https://user-images.githubusercontent.com/5360142/62254195-a5ee3f00-b3ce-11e9-95e4-f34fcedcac02.png">

that really helps.

And also, "when opening the help file" doesn't really isolate the problem. Help file is big. Try to find and isolate the problem (which object seems to be giving this error?). 

Anyway, if you're using the "read" message and getting the same error, how about testing that [fromsymbol] object? What do you find?

**Comment by porres** [comment]
About the buffer size error. This is given if you set a buffer size of "0" to it. Which is not the case... because you have [cyclone/buffer~ tom_$0 -1 2], so you're giving it a size of "-1", which does not trigger this error

<img width="716" alt="Screen Shot 2019-07-31 at 20 14 33" src="https://user-images.githubusercontent.com/5360142/62254628-485af200-b3d0-11e9-8ca3-bde99b91b5b9.png">

**Comment by porres** [comment]
the "array: couldn't find" error can only be given by the "crop" message, as it's the only place we have [array] objects being used.

<img width="360" alt="Screen Shot 2019-07-31 at 20 32 16" src="https://user-images.githubusercontent.com/5360142/62255158-69bcdd80-b3d2-11e9-80ad-8b6e1cd90270.png">


So it really makes no sense that you're getting these errors just by loading the object and turning the DSP on.

**Comment by tommyecho** [comment]
The fromsymbol object seems to work fine, but I’m positive it has to be something that isn’t getting passed from some object. 

It does strike me as being strange that the message says the buffer size isn’t defined, since the setsize message is being received and should resolve that issue?

This message was sent from my 
Phone.

**Comment by tommyecho** [comment]
I agree that it doesn’t make sense

This message was sent from my 
Phone.

**Comment by porres** [comment]
> It does strike me as being strange that the message says the buffer size isn’t defined, since the setsize message is being received and should resolve that issue?

I don't know, I'm really confused, you seem to be talking to me about a patch I don't see. What set message are you talking about? Being received how and why??? 

Again, just try to send me a patch, with as little things as possible, isolating the issue, telling me what is wrong with it, and we'll debug. I'm sorry, but this is getting really confusing and not quite objective to me. Help me help you :)

**Comment by tommyecho** [comment]
Ok, I hope this sends and hope it clarifies. I had mentioned this part of my troubleshooting in a previous response, but I wasn’t clear enough.  Just let me know if you don’t see the attached zip file and I’ll upload it to google drive



Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
> Just let me know if you don’t see the attached zip file and I’ll upload it to google drive

I see nothing :(

**Comment by tommyecho** [comment]
Sorry about that.

Here’s the link:

https://drive.google.com/open?id=1HEL3VpoYquKEnDjlisPfNdLj0M9h1Lnb


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
ok, I have the patch, so 

- When setting the size, it usually freezes? Or just when turning the DSP on it freezes?
- You already have the buffer size set to "1000" ms with the arguments. Why the set message then?
- Does it freeze and print the warings without using the set message? Just by having the object with the given arguments setting the size to 1000ms?

- Cause it seems that the set message is just irrelevant here, the thing is that the warnings are given even though the buffer has a size that was set! If you gave it "1000" as the argument, the array size was in fact set to 1000 ms, right?

- Cause then we can forget about the setsize message, I don't understand what's the deal with it, we can simplify the patch!

- Now, note the following, the error should be given when the object is created and the arguments are loaded. Here's the only way this error can be given and triggered, when you give it a size of "0" as an argument...

<img width="291" alt="Screen Shot 2019-08-01 at 02 19 27" src="https://user-images.githubusercontent.com/5360142/62267572-e7004680-b402-11e9-9434-2abf86570a92.png">

So, again, the question is, if you did not give it a "0" size as an argument, and you gave it anything else, like "1000" and it works, how on earth does it give you this error when you turn on the DSP?

It makes no sense!  Go ahead, check the buffer~.pd file.

The warning is given only inside [pd load-args], It should be given only once, at the time [loadbang] sends a bang... nothing of this makes sense...

if you have [cyclone/buffer~ tom_$0 1000], the 1st argument is sent to [s $0-name] as the buffer name  in [pd buffer_arg], and in [pd 2nd_arg], the 2nd argument is converted to samples with [cyclone/mstosamps~], and if it's not "0", the error warning cannot be given... not then, not ever after then...

I have no idea what and how this is happening...

**Comment by porres** [comment]
> When setting the size, it usually freezes? Or just when turning the DSP on it freezes?

I mean, when turning the DSP on in general it freezes, or just when using the setsize message? I think just whenever the DSP is on, right?

**Comment by porres** [comment]
So, just create an object with [buffer~ tom_$0 1000]

- tell me what happens when the patch is opened/created
- tell me what happens when you turn on the DSP

**Comment by porres** [comment]
moreover, what happened to the other error messages? like the 'array' error messages? 

With this last patch that you sent me, what are you getting? Just print messages that say "Warning: [buffer~]'s size not defined".? Nothing else?

as you can seem things are still very confusing, there is just too much going on and the information keeps changing and changing... 

so let's focus!

First let's debung the patch with just [buffer~ tom_$0 1000]...

- tell me what happens when the patch is opened/created
- also tell me if you can cofirm if the given size was successfully set
- and tell me what happens when you turn on the DSP.

IF nothing is printed when you create the oject or load the patch, if the table size is correctly set. Then the patch works, there's nothing wrong with it.

If lots of errors start showing up only when you turn on the DSP, I'm sorry... seems impossible!

See if you casn share the patch with others in the facebook group and if they can replicate the issue.

**Comment by porres** [comment]
and it's confusing because I don't knoe now how the other errors asre being givem with the $2 out of rangem and the array error... you'd need to give me separate example patches to explain how they happen... but it just seems weird as hell as the other one...

**Comment by porres** [comment]
you can also try [else/sample~], which is a similar object I coded. 

The ELSE library is available via deken as well. Please check this object and tell me if it's crazy too... it should be very much about the same one...

**Comment by tommyecho** [comment]
I appreciate your help with this. Sorry it’s confusing. I thought I was been clear as to what happens when. The main problem is this warning depicted in this video. As you can see, no message is printed at the moment the object is created. As soon as I click dsp 1, the messages start printing really rapidly. Most of the time I have to shut down because it freezes 


This message was sent from my 
Phone.

**Comment by tommyecho** [comment]
I’d like to focus on this error that is in the video. It’s the main thing. It’s what causes crashing. After that, it may not work until I figure out what isn’t getting the $2 variable, but that’s another thing that only happens when I send a “read” message to the buffer to have it read a file. So yes let’s focus on the video

I hope that video works so that you don’t think I’m trolling you. I appreciate the help

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
> The main problem is this warning depicted in this video

which "video"?

**Comment by tommyecho** [comment]
arg. I don’t know why nothing I send as an attachment shows up on your end. I’m reasonably computer literate and never have this problem ha

Ok I’ll google drive the video. just a sec

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
> I hope that video works so that you don’t think I’m trolling you. I appreciate the help

a video helps! But it wasn't uploaded, it seems whatever you're doing to send me messages via your phone is not working. Perhaps is better to use the github framework to exchange messages and attach links/files/images

**Comment by tommyecho** [comment]
I am not savvy with github, but the other attachments were sent from my computer and they didn’t work either. The video is uploading to google drive now

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
access this link: https://github.com/porres/pd-cyclone/issues/451

reply from there

**Comment by tommyecho** [comment]
The video is still uploading. All that it is is proof that this is happening:

1 I make the object [cyclone/buffer~ tom_$0 1000] and nothing is printed in the pd printer

2 I turn on the DSP and many many messages are printed, all saying the same thing: “Warning: buffer’s size not defined”

3. I turn off the DSP and the messages stop printing

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by tommyecho** [comment]
Here’s the vid:

https://drive.google.com/file/d/1CK1PJ48XQmt2QRq2J-EIrCKwA9rFJqCp/view?usp=sharing


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by tommyecho** [comment]
else/sample~ seems to work fine

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
I still think you're trolling, haha, 

un-fuckin-believable

and after you created the object, you can verify that the array size was actually set to "1000" ms, right?

**Comment by porres** [comment]
please try with this object instead... I now have numbered all the warnings. Please tell me which warning number(s) appear.

[buffer~.pd.zip](https://github.com/porres/pd-cyclone/files/3455123/buffer.pd.zip)

**Comment by tommyecho** [comment]
Yeah, it’s fuckin nuts!

Ok this is a good question. In past tests, I was verifying the array size after sending it the setsize message, which does change the array size. It seems that the argument of 1000 is unrecognized. Weirdly, the array size is 100, not the expected 44100. I tried it again and same result.

Now, If I turn on DSP, then send it the setsize 1000 messge, then manage to get the DSP to turn off (instead of freezing), the array is now 44100. This is the way it’s been. Your question made me get more specific about it though. Trying to be objective here

**Comment by porres** [comment]
>  It seems that the argument of 1000 is unrecognized. Weirdly, the array size is 100, not the expected 44100. I tried it again and same result.

this is bad!

**Comment by tommyecho** [comment]
It is. I’ve been working on this one problem for countless hours lol

Ok I downloaded that zip onto my raspberry pi, but I don’t know how to use it:/

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
> Ok I downloaded that zip onto my raspberry pi, but I don’t know how to use it:/

can't you unpack the zip?

**Comment by tommyecho** [comment]
do I drop it in the extras folder? usr/lib/pd/extra/cyclone

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
> do I drop it in the extras folder? usr/lib/pd/extra/cyclone

yup, unpack it, get this new buffer~.pd file and put it in there to test it

**Comment by porres** [comment]
I wanna know if we seem "Warning1" / "Warning2" or "Warning3"

**Comment by tommyecho** [comment]
Well I extracted it, but then I double click on the file and pure data opens up and gives a really bizarre message in the print window: weird symbols and then “no such object”

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
yikes... it's supposed to be a replacement of the buffer~ object...

anyway, you could do it yourself... see all the [print] objects in your buffer~ file and give tem numbers to tell you which one is being called. This warning can come from 3 different places, so we need to know which one

**Comment by porres** [comment]
If no buffer size is given, the array size should be "1", not "100"...

but "100" is what the array size is when you create it in Pd...

nothing makes sense...

**Comment by porres** [comment]
I think it comes from the print inside [pd 2nd_arg]

**Comment by porres** [comment]
we'll give this a bit more attention, but the next step is that you upload this patch to the Pd group or something and ask other people to see if they can replicate the issue!

**Comment by tommyecho** [comment]
Ok! I’m searching for print objects. I’ve found one and it is in 2nd args. it already has a label “Warning” ha

is there more than one print object called “warning”? if not, then this has to be the one
Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by tommyecho** [comment]
Ok, I see. It looks like one warning messge is coming from arg 3.  Still looking 
Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
what I had done was name them "Warning1" / "Warning2" / "Warning3"

**Comment by tommyecho** [comment]
Yeah I named them warningbuff warning2 and warning3

I’m getting warning3 and I’m also getting just warning, which seems impossible. I went through the whole thing and only saw those three print objects

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
> I’m getting warning3

and where is that, which subpatch?

> I’m also getting just warning, which seems impossible.

many impossible things are happening 
> I went through the whole thing and only saw those three print objects

There are only 3 print objects that give the  "Warning: [buffer~]'s size not defined"

if you've named them all, what you're seeing just can't be... when are you admitting you're trolling me? :)

**Comment by tommyecho** [comment]
hahaha

I knew it must be impossible. I swear it’s true though. I even reentered the names in the print objects

so it’s the 3rd arg subpatcher that is the warning 3


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by tommyecho** [comment]
oooooohhhh I found the 4th warning print

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
>  I swear it’s true though. I even reentered the names in the print objects

it really seems you're missing something obvious, I can't see from distance

what's left to do now is share the object, the example and see if people can replicate your issue!

you may also try the Pd list, use the videos and keep things quite objective ;)

**Comment by tommyecho** [comment]
now I think you’re trolling me. . .

it’s inside channels. theres a subpatcher called pd attributes. Therein lie the 4th print object. I labeled it warning4 and that was it. so it’s warning 3 in arg3 and warning4 in attributes

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by tommyecho** [comment]
I’ll post some online tomorrow about this to try to get some folks to replicate the behavior, but those are the warning messages, at least


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
> now I think you’re trolling me. . .

I didn't, you can count, look again...

There are only 3 print objects that give the "Warning: [buffer~]'s size not defined" message

you may have numbered another 4th [print] object that was giving another message

**Comment by tommyecho** [comment]
huh, well I just tagged on numbers on the 4 warning words. seriously

anyways, they’re identified now. i’m pretty mystified by this


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
4 warning words, but only 3 of them are warnings on "[buffer~]'s size not defined"

**Comment by tommyecho** [comment]
oh I see. the first warning object was getting a message for buffer name not given.

ok the one that I labeled warning 3 is in 3rd args still and the one I labelled warning4 is in attributes still


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
"1000" ms is sent to the [2nd arg] subpatch

it should become 44100 in here 

<img width="269" alt="Screen Shot 2019-08-01 at 04 07 44" src="https://user-images.githubusercontent.com/5360142/62272690-0357af80-b412-11e9-9667-8bc39ad052fd.png">

can you test it?

**Comment by porres** [comment]
so, you should do some debugging yourself now...

move up the chain, get prints from above... you know where the messages are being sent to print now, so next, connect, for instance, the output of the [select 0] object, just to see what comes out of it (should be bangs, of course). Then up with [cyclone/mstosamps~], [select -1] and so on...

that's the art of debugging

we have to find how and where this print message is getting triggered

**Comment by tommyecho** [comment]
Oh yeah I know how to debug; I’ve used Max for a long time. this linux and pd stuff is a little new to me still. Especially the Linux side of it. 

I didn’t realize those warning messages were coming from print objects within your patch. I was thinking maybe it was something internal to pd. Now that I have this, I think I can get some real progress done!

ok I did that quick test and the 2nd argument does work. the print object showed the expected results.

Earlier in this, I went through the resize subpatch with print objects and everything seems to work fine in there. 

This has been extraordinarily helpful. Thanks for your time. I’ll get back with this tomorrow. Hopefully I can figure out the rest on my own, but let me know if anything comes to mind that I should do in particular

**Comment by tommyecho** [comment]
Here’s something I just discovered that I think might be causing some of the troubles. On the raspberry pi, the mstosamples~ object continuously and constantly sends it’s last output number (as long as the DSP is on). 

I did a comparison with how this object behaves on my mac, and it sends the number just once. For example, I send 2000 to its inlet and I get 88200 out its right outlet only once in the print screen. In raspbian, I get a stream of 88200 in the print window

I’m imagining this is not what should be happening, right?

**Comment by porres** [comment]
> I’m imagining this is not what should be happening, right?

yeah, bug detected

**Comment by porres** [comment]
> I did a comparison with how this object behaves on my mac, and it sends the number just once. For example, I send 2000 to its inlet and I get 88200 out its right outlet only once in the print screen. In raspbian, I get a stream of 88200 in the print window

Ok, but in order for things to make sense, what shoud be happening is that [cyclone/mstosamps~] needs to be outputing "0" even if it never received any input. Can you confirm that?

**Comment by porres** [comment]
anyway, in the meantime, I suggest you use the [else/sample~] object instead. It should do everything you want and need

**Comment by tommyecho** [comment]
Nope, It’s sending 88200 over and over again. I’ll continue to investigate in a couple of hours here. I think I’m getting closer. 

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
> Nope, It’s sending 88200 over and over again.

what do you mean?

Here's the test I want you to do. Created [cyclone/mstosamps~], connect to a [print] in the float output. Don't send it any message, just turn on the DSP and see if it outputs "0" over and over...

**Comment by tommyecho** [comment]
Oh I see! I thought you were talking about the mstosamps~ within the arg2 subpatch.

Ok I just ran that test and mstosamps does indeed send continuous 0’s


Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
how about testing other earlier versions of cyclone, can you do that?

here's 0.2 https://puredata.info/Members/fjkraan/software/cyclone/0.2beta1/

**Comment by porres** [comment]
direct link for the pi

https://puredata.info/Members/fjkraan/software/cyclone/0.2beta1/cyclone-v0.2beta1-%28Linux-armv7l-32%29%28Sources%29-externals.tgz/at_download/file

**Comment by porres** [comment]
and please, see if you also get the same unexpected behaviour with [cyclone/sampstoms~]

**Comment by tommyecho** [comment]
I’m just a bit wary of installing an older version, since it was a lot of work to get my pi updated. Isn’t that the version that comes up in the external search for cyclone when running PD .47? If so, then I think it’s the same.

I can try it out though. gotta figure out how to install it this way.

I got the error messages to stop happening. I’ll try to explain what I did as clearly as I can:

I put a send object in 2nd argument patch to send the size number to the  mstosamps~ in arg 3 and attributes subpatches. This stopped the error messages from happening, because now all the mstosamps objects are loaded with the size.

Please let me know if you think this is a sufficient solution to get the buffer~ to work on raspbian. If so, then here is the next issue

 I still can’t get the record~ object to record anything into the buffer. I get the message “record’s no such array ‘0-tom_1005’  Any ideas about that?

Thanks!

**Comment by porres** [comment]
> Please let me know if you think this is a sufficient solution to get the buffer~ to work on raspbian

I don't :) the solution would be to fix mstosamps~

if you want to get your patch working, please try using the else/sample~ object instead

 
> I still can’t get the record~ object to record anything into the buffer. I get the message “record’s no such array ‘0-tom_1005’  Any ideas about that?

None. My ideas is that all objects work fine :) and I'd need a better report.

> I’m just a bit wary of installing an older version, since it was a lot of work to get my pi updated. 

swapping library versions are as trivial as switching folders. Not to worry.

> Isn’t that the version that comes up in the external search for cyclone when running PD .47? If so, then I think it’s the same.

I don't know, are you saying you were using cyclone 0.2 before?

Anyway, now I need you to make sure you're using  mstosamps~ from cyclone 0.3, and that this is the one that misbehaves...

also please make sure how the previous 0.2 version works

let me go ahead and tell you that Derek Kwan, also a maintainer of cyclone, did test  mstosamps~ in a raspberry pi 3, with cyclone 0.3 and vanilla 0.49 and found no issues... people on facebook were also saying they did not have a problem with buffer~ on their system, so it just seems your issue is not being replicated...

![67576305_519616935445583_7323259070377361408_n](https://user-images.githubusercontent.com/5360142/62329581-bbc23980-b48c-11e9-94f4-ad1743da9e92.jpg)

**Comment by porres** [comment]
Derek also tested with cyclone 0.2 and found no issue...

I really don't know what to tell you

seems like something crazy on your system

burn it all down, nuke it all, reinstall everything from the ground up? 

ask on the pd-list/pd-dev for help? Cause this does not seem like a cyclone issue...

**Comment by tommyecho** [comment]
arg ok I will be burning everything and trying to rebuild.

I am running cyclone .3. A couple of weeks ago I was on .2 because my puredata was at .47 then

thanks so much for the help. I’m working with a patch with else/sample~ and it’s working fine. I definitely will be rebooting from the ground up though

Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
if things are still funny, we'll see what can be done. One way or another, I care about Pd functioning properly, so if this is a Pd bug and not a cyclone issue, let's try and fix it

**Comment by porres** [comment]
by removing everything and start over, I mean like getting a new SD card and everything, really from ground zero...

**Comment by tommyecho** [comment]
Yep I’m gonna do that 

This message was sent from my 
Phone.

**Comment by porres** [comment]
you can also try for help and see what other pi users can report/replicate at https://forum.pdpatchrepo.info/ and the pd-list

**Comment by tommyecho** [comment]
Hey Alexandre,

I just now got everything up and running, after erasing my SD and setting up a new system. I am not having any problems with the buffer~, it works great! Also, the mstosamps~ object works as it should, sending the number only once

I’m really sorry to have been so much trouble. If it makes any difference, I spent a whole lot of time on this by myself before reaching out to you about it. I do greatly appreciate your time in helping me with it. It’s been a learning experience. I have no idea why the those two objects were not working in particular, but the problem is completely resolved. I have some issue with priority scheduling not being quite right, but that’s a completely different issue that I’ll do some research on and figure it out.

Thanks so much for your patience and help,

Tom

**Comment by porres** [comment]
>  I have no idea why the those two objects were not working in particular, but the problem is completely resolved. 

So you won't tell the truth that you were trolling me, huh?

haha, anyway, there can't be any explanation... I mean, there is, but it's way over our heads :)

**Comment by tommyecho** [comment]
Hahaha it is a mystery 

This message was sent from my 
Phone.

**Comment by tommyecho** [comment]
Is bucket missing from cyclone v 0.3 for linux? I had it running in some patches using earlier versions of cylone a while back, but it’s not showing up now. . .
Dr. Thomas Echols
Classical Guitarist
www.thomasechols.com
Composer, Songwriter
www.manwomanfriendcomputer.com

Listen to Plainte Calme <https://itunes.apple.com/us/album/plainte-calme/id399173764> on iTunes and
Man, Woman, Friend, Computer <http://manwomanfriendcomputer.bandcamp.com/album/man-woman-friend-computer> on Bandcamp

**Comment by porres** [comment]
opening a new issue for it https://github.com/porres/pd-cyclone/issues/453

**Comment by porres** [comment]
please mention there on the new issue where you did not find the object, raspberry pi or what?
